lib/provisioning/Provisioner.js

```diff
@@ -129,24 +143,170 @@ Provisioner.prototype._validate = function(actual, parameterName) {
 };
 
 // Validate parameters for use in linking/unlinking
-Provisioner.prototype._validateAll = function(parameters) {
-    let parameterNames = ['matrix_room_id', 'remote_room_channel', 'remote_room_server'];
+Provisioner.prototype._validateAll = function(parameters, parameterNames) {
+    if (!parameterNames) {
+        parameterNames = [
+            'matrix_room_id',
+            'remote_room_channel',
+            'remote_room_server',
+            'op_nick',
+            'key'
+        ];
+    }
     for (var i = 0; i < parameterNames.length; i++) {
         this._validate(parameters[parameterNames[i]], parameterNames[i]);
     }
 };
 
+Provisioner.prototype._updateBridgingState = Promise.coroutine(
+    function*(roomId, bridger, status, skey) {
+        let intent = this._ircBridge.getAppServiceBridge().getIntent();
+        try {
+            yield intent.client.sendStateEvent(roomId, 'm.room.bridging', {
+                bridger: bridger,
+                status: status // pending, success, failure
+            }, skey);
+        }
+        catch (err) {
+            throw new Error(`Could not update m.room.bridging state in this room ($err.message)`);
+        }
+    }
+);
+
+Provisioner.prototype._authoriseProvisioning = Promise.coroutine(
+    function*(server, bridger, ircChannel, roomId, opNick, key) {
+        let ircDomain = server.domain;
+
+        // (Matrix) Check power level of user
+        let matrixClient = this._ircBridge.getAppServiceBridge().getClientFactory().getClientAs();
+        var lookup = new StateLookup({
+            client : matrixClient,
+            eventTypes: ['m.room.power_levels']
+        });
+
+        yield lookup.trackRoom(roomId);
```

Comment:
You don't need to be tracking any updates to the `m.room.power_levels` event you receive, so just use [getStateEvent](http://matrix-org.github.io/matrix-js-sdk/0.5.5/module-client-MatrixClient.html#getStateEvent).